packages/wrangler/src/dev/proxy.ts

```diff
@@ -686,7 +686,7 @@ export async function waitForPortToBeAvailable(
 					doReject(err);
 				}
 			});
-			server.listen(port, () =>
+			server.listen(port, "localhost", () =>
```

Comment:
In this file the user may actually specify an IP address on which the proxy is to listen. I think it might be safest to use this IP address when waiting for a port to be available.
I think that it is possible for a port on localhost to be free while the same port on another interface is blocked, right?

Reply:
So in general we could pass the IP address via the `waitForPortToBeAvailable()` call to the `checkPort()` call.

Reply:
Done, I'll test this with `wrangler dev --remote` when pre-release builds are available.